### DISCUSSION TITLE
Docker support

### DISCUSSION THREAD

**Discussion by matthew73210** [Nachtzuster/BirdNET-Pi#14] [Ideas]
Hey,
As the old version is starting to die, i'll probs move ship here. Currently I don't have it installed as my orange pi is messing about. 
I was wondering though if you had thought about getting this running on docker? I'm in the process (2 month haitus rn because of work) of building my own version, no front end. Just the analyser and a rtsp recorder --> db or mqtt stream. Using the S6 overlay (easy service / deamon stuff). But if i didn't have to do all that and just do a docker pull on my server would be ace. 
What are your thoughts? Might be a hassle but going from systemd to S6 is a pain at best.

**Comment by Nachtzuster** [comment]
No, no docker plans here.
People have been running it in a VM to get the isolation, I believe.

**Comment by Vertikar** [comment]
Did you get any further with this @matthew73210 ?
I'd really like to run Birdnet-PI alongside Birdnet-go and having both running in docker seems like the cleanest way to do  this.

**Comment by alexbelgium** [comment]
Hi, I have made homeassistant addons for both here  : https://github.com/Nachtzuster/BirdNET-Pi?tab=readme-ov-file#homeassistant-addon. That means we can actually run them as docker containers in parallel in they same system listening on the same rtsp feed.

However it is a container tailored to homeassistant which is not universal (this would require just some adaptation). You might try to spin a VM with HA to try it : it is really worth it. There is just so many options in birdnet pi and the owner of this fork has done a great work to upgrade and update things.

**Reply by Vertikar** [reply]
Thanks Alex, am already running HA, but on docker, rather than HAOS unfortunately.

I was previously running BirdNET-Pi on its own, until the SD card died and thought I'd have a play with the Go version, but I'm definitely missing the reporting, notifications and other features.

I'll have a dig through your dockerfile and see if I can build it myself

**Reply by alexbelgium** [reply]
well if you are on HA supervised I think you can install addons ! HAos is not a requirement

We can look together at the dockerfile perhaps we can even use the current one but modify it to support both HA and docker

**Reply by alexbelgium** [reply]
I've started a specific section in the dockerfile https://github.com/alexbelgium/hassio-addons/blob/master/birdnet-pi/Dockerfile to add non-HA speciifc elements. Then people would just need to create a /data/options.json file that would be equivalent to what HA would generate, and it should be usable indepedently of HA
